src/l2/XERC20PufferVault.sol

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.4 <0.9.0;
+
+import { IXERC20 } from "./interface/IXERC20.sol";
+import { UUPSUpgradeable } from "@openzeppelin-contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import { AccessManagedUpgradeable } from
+    "@openzeppelin-contracts-upgradeable/access/manager/AccessManagedUpgradeable.sol";
+import { ERC20PermitUpgradeable } from
+    "@openzeppelin-contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+
+contract XERC20PufferVault is IXERC20, AccessManagedUpgradeable, ERC20PermitUpgradeable, UUPSUpgradeable {
+    /**
+     * @notice The duration it takes for the limits to fully replenish
+     */
+    uint256 private constant _DURATION = 1 days;
```

Comment:
Should we use [ERC-7201: Namespaced Storage Layout](https://eips.ethereum.org/EIPS/eip-7201) as we did in our mainnet contracts?

Reply:
I'd change this to use ERC7001, we are using that pattern in all of our upgradeable contracts, and this will be one of them, so we should be consistent, even if it means modifying the official implementation a bit